src/java/org/apache/cassandra/db/compaction/unified/Controller.java

```diff
@@ -420,46 +420,49 @@ public void onStrategyBackgroundTaskRequest()
      * length of data on disk.
      *
      * @param length the total length on disk
-     * @param W the scaling parameter to use for the calculation
+     * @param scalingParameter the scaling parameter to use for the calculation
      *
      * @return the read amplification of all the buckets needed to cover the total length
      */
-    public int RA(long length, int W)
+    public int readAmplification(long length, int scalingParameter)
     {
         double o = getSurvivalFactor();
         long m = getMinSstableSizeBytes();
 
-        int F = W < 0 ? 2 - W : 2 + W;
-        int T = W < 0 ? 2 : F;
-        int bucketIndex = Math.max(0, (int) Math.floor((Math.log(length) - Math.log(m)) / (Math.log(F) - Math.log(o))));
+        int F = scalingParameter < 0 ? 2 - scalingParameter : 2 + scalingParameter;
+        int T = scalingParameter < 0 ? 2 : F;
+        int maxIndex = Math.max(0, (int) Math.floor((Math.log(length) - Math.log(m)) / (Math.log(F) - Math.log(o))));
 
         int ret = 0;
-        for (int i = 0; i < bucketIndex; i++)
+        for (int i = 0; i < maxIndex; i++)
             ret += T - 1;
 
+        if (scalingParameter >= 0)
```

Comment:
We should also add 1 in an `else` case here (`maxIndex` is inclusive).

Reply:
b0d596ac1d055d975503e55ddcf0b79ddf7bf770

Reply:
Looks like there already was an `else` case that I've missed. Apologies.

Reply:
41de74e52d9184a0982e38c465f1b29a4a6a300e should fix amplification estimation. Please note that I also removed `- 1` from leveled write amplification to match `+1` we add in the tiered case.